File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max_sum = INT_MIN, cur_sum = 0;
        for(int num : nums) {
            cur_sum += num;
            max_sum = max(cur_sum,max_sum);
            if(cur_sum < 0) cur_sum = 0;
        }
        return max_sum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max_sum = Integer.MIN_VALUE;
        int sum = 0;
        for(int num : nums) {
            sum += num;
            max_sum = Math.max(max_sum,sum);
            if(sum < 0) sum = 0;
        }
        return max_sum;
    }
}